### PR TITLE
feat(codegen): print linked and external legal comment

### PR DESCRIPTION
**Pull request by Boshen**
part of #7050

### CODE REVIEW

**Comment by Boshen** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/7059?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#7060** <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/7060?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#7059** <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/7059?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#7058** <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/7058?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#7054** <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/7054?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @Boshen and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

**Review by Dunqing — CHANGES_REQUESTED** [review]
Seems we don't need to handle unsed_comments in `print_comments`

File: crates/oxc_codegen/src/comment.rs [hunk 1]
```diff
@@ -89,11 +78,36 @@ impl<'a> Codegen<'a> {
...
+                        self.legal_comments.push(comment);
+                        continue;
+                    }
+                    LegalComment::None => {}
+                }
+            }
+            unused_comments.push(comment);
+        }
+
+        self.print_comments(start, &leading_comments, unused_comments);
```
Comment (Dunqing) [inline crates/oxc_codegen/src/comment.rs:110]: Seems we don't need to handle `unused_comments` in `print_comments`

**Review by Dunqing — APPROVED** [review]
LGTM

File: crates/oxc_codegen/src/comment.rs (hunk 1 above)
Comment (Dunqing) [reply crates/oxc_codegen/src/comment.rs:110]: We can change it in a follow-up PR if you think the same as me.

**Pull request merged**